lib/i18n/tasks/translators/deepl_translator.rb

```diff
@@ -60,21 +60,102 @@ def options_for_plain
 
     # @param [String] value
     # @return [String] 'hello, %{name}' => 'hello, <i18n>%{name}</i18n>'
-    def replace_interpolations(value)
+    def original_replace_interpolations(value)
       value.gsub(INTERPOLATION_KEY_RE, '<i18n>\0</i18n>')
     end
 
     # @param [String] untranslated
     # @param [String] translated
     # @return [String] 'hello, <i18n>%{name}</i18n>' => 'hello, %{name}'
-    def restore_interpolations(untranslated, translated)
+    def original_restore_interpolations(untranslated, translated)
       return translated if untranslated !~ INTERPOLATION_KEY_RE
 
       translated.gsub(%r{</?i18n>}, '')
     rescue StandardError => e
       raise_interpolation_error(untranslated, translated, e)
     end
 
+    # deepl does a better job with interpolations when it doesn't have to deal
+    # with <br> tags, so we replace all of them with meaningless asterisk chains
+    BR_REGEXP = %r{(<br\s*/?>\s*)+}i.freeze
+    BR_SINGLE_MARKER = ' *** '
+    BR_DOUBLE_MARKER = ' ***** '
+
+    # letting deepl 'read' the interpolations gives better translations (and
+    # solves the problem of interpolations getting pushed all the way to the
+    # front of the sentence), however, deepl will also try to translate
+    # the interpolations and that gets messy.
+    # we use nonsense three-letter acronyms so deepl will 'read' them and leave
+    # them alone (the letter X also works very well, except in sentences with
+    # several consecutive interpolations because that reads X X X and deepl
+    # doesn't handle that well)
+    # deepl also needs to know if an interpolation will be a word or a number,
+    # for romance languages it matters. a little Spanish lesson to illustrate:
+    # "%{foo} betalingen" translates either to "facturas %{foo}"
+    # (openstaande betalingen -> facturas pendientes) or to "%{foo} facturas"
+    # (5 betalingen -> 5 facturas)
+    # for interpolation keys that are usually numeric, we pick a number
+    # instead of the three-letter acronym (more consistency in how we name
+    # interpolation keys would help)
+    LETTER_SUBS = %w[RYX QFN VLB XOG DWP ZMQ JZQ WVS LRX HPM].freeze
+    NUM_SUBS = %w[17 19 23 29 31 37 41 43 47 53].freeze
+
+    def key_substitution(stache, index)
+      case stache.gsub(/[^a-z]/, '')
+      when 'count', 'minutes', 'hours'
+        NUM_SUBS[index % NUM_SUBS.size]
+      else
+        LETTER_SUBS[index % LETTER_SUBS.size]
+      end
+    end
+
+    # BEX version of replace_interpolation
+    def replace_interpolations(value)
+      key_index = 0
+      value.gsub(INTERPOLATION_KEY_RE) do |stache|
+        sub = key_substitution(stache, key_index)
+        key_index += 1
+        "<var stache=\"#{stache}\" sub=\"#{sub}\">#{sub}</var>"
```

Comment:
Why use 'var'? We should either keep using i18n or add 'var' to the ignore tags: https://github.com/bookingexperts/i18n-tasks/blob/383db5d48a1b5df8192cc8beca80d8687e88345d/lib/i18n/tasks/translators/deepl_translator.rb#L50 . It not being in the ignore tags may also explain the many cases needed in the 'restore_interpolations' function

Reply:
@tom-brouwer-bex I do not believe there's a way to pass these tests with an ignored tag
see the Spanish lesson in the comment, X taken can not be translated correctly without a hint about what X might be. It's either remove i18n from the ignored tags list (which I think would be confusing) or use a different tag

Reply:
Ah ok. I agree, if your tests indicate that this is the key to getting proper substitutions, then I guess it's a feature, not a bug 😄